onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -112,22 +211,62 @@ def get_intermediate_outputs(self, calib_mode='naive'):
         '''
 
         #conduct inference session and get intermediate outputs
-        session = onnxruntime.InferenceSession(self.augmented_model_path, None)
-
-        intermediate_outputs = []
-        while True:
-            inputs = self.data_reader.get_next()
-            if not inputs:
-                break
-            intermediate_outputs.append(session.run(None, inputs))
-        node_output_names = [session.get_outputs()[i].name for i in range(len(intermediate_outputs[0]))]
-        output_dicts_list = [
-            dict(zip(node_output_names, intermediate_output)) for intermediate_output in intermediate_outputs
-        ]
+        if providers:
+            sess_options = onnxruntime.SessionOptions()
+            sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL #ORT_ENABLE_BASIC
+            session = onnxruntime.InferenceSession(self.augmented_model_path, sess_options=sess_options, providers=providers)
+        else:
+            session = onnxruntime.InferenceSession(self.augmented_model_path, None)
 
         #number of outputs in original model
         model = onnx.load(self.model_path)
         num_model_outputs = len(model.graph.output)
+        model_original_outputs = set(output.name for output in model.graph.output)
```

Comment:
you should be able to get it with session.get_outputs()

Reply:
I think here it's to get original model's outputs not the augmented model's outputs.